platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -253,29 +255,44 @@ public ProjectLayoutSetupRegistry getProjectLayoutRegistry() {
     @TaskAction
     public void setupProjectLayout() {
         UserInputHandler inputHandler = getEffectiveInputHandler();
+        GenerationSettings settings = inputHandler.askUser(this::calculateGenerationSettings).get();
+
+        boolean userInterrupted = inputHandler.interrupted();
+        if (userInterrupted) {
+            throw new BuildCancelledException();
+        }
+
+        settings.getInitializer().generate(settings.getSettings());
+        generateWrapper();
+
+        settings.getInitializer().getFurtherReading(settings.getSettings())
+            .ifPresent(link -> getLogger().lifecycle(link));
+    }
+
+    private GenerationSettings calculateGenerationSettings(UserQuestions userQuestions) {
```

Comment:
💅 Alternate method name ideas: `surveyUser`, `gatherGenerationSettings`, `configureGeneration`.

Like 265 `settings.getSettings()` doesn't read great.  Maybe `GenerationSettings` could be renamed to `GenerationConfig`?  Or maybe it's `InitSettings` that ought to be renamed to `InitSpec` or something.